analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R

```diff
@@ -0,0 +1,373 @@
+# This script performs the following functions:
+# 1. DESeq2 tumor-only analysis with RUVg by molecular subtype
+# Authors: Komal Rathi, updated by Adam Kraya
+
+suppressPackageStartupMessages({
+  library(optparse)
+  library(tidyverse)
+  library(DESeq2)
+  library(RUVSeq)
+  library(EDASeq)
+  library(edgeR)
+  library(stringr)
+  library(data.table)
+})
+
+# parse parameters
+option_list <- list(
+  make_option(
+    opt_str = "--dataset",
+    type = "character",
+    help = "Give any name to the dataset to create output folder"
+  ),
+  make_option(
+    opt_str = "--cancer_group_values",
+    type = "character",
+    help = "cancer group value"
+  ),
+  make_option(
+    opt_str = "--cohort_values",
+    type = "character",
+    help = "cohort value"
+  ),
+  make_option(
+    opt_str = "--k_value",
+    type = "character",
+    help = "number of K values to evaluate"
+  ),
+  make_option(
+    opt_str = "--pos_c",
+    type = "character",
+    help = "path to file of positive control genes"
+  ),
+  make_option(
+    opt_str = "--neg_c",
+    type = "character",
+    help = "path to file of negative control genes"
+  )
+)
+
+# extract parameters
+opt <- parse_args(OptionParser(option_list = option_list))
+dataset <- opt$dataset
+cancer_group_values <-
+  unlist(stringr::str_split(opt$cancer_group_values, ','))
+cohort_values <- opt$cohort_values
+k_value <- as.numeric(opt$k_value)
+neg_ctrl_genes.f <- opt$neg_c
+pos_ctrl_genes.f <- opt$pos_c
+
+# establish directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analysis_dir <-
+  file.path(root_dir, 'analyses', 'rnaseq-batch-correct')
+input_dir <- file.path(analysis_dir, 'input')
+plots_dir <- file.path(analysis_dir, 'plots', dataset)
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+
+output_dir <- file.path(analysis_dir, 'output', dataset)
+if (!dir.exists(output_dir)) {
+  dir.create(output_dir, recursive = TRUE)
+}
+
+# source functions
+source('util/deseq2_pvals_histogram.R') # DESeq2 pval histograms
+source('util/edaseq_plot.R') # PCA and UMAP clustering
+source('util/ruvg_test.R') # function to run RUVg
+
+# read histology
+message("Reading in histologies file")
+htl_df <- readr::read_tsv('../../data/v10/histologies.tsv')
```

Comment:
```suggestion
htl_df <- read_tsv(file.path(data_dir, "histologies.tsv"), guess_max = 100000)
```

Can you change all of the paths to use file.path() and use the current data directory (which also should be added above) without direct links to a version?

Reply:
yes. 
currently, it is written to compare tumor-only based on molecular subtype. since MYCN subtypes were migrated to pathology free text for NBL, I hardcoded temporarily for v10 but will change as described above once end-to-end testing by reviewers is complete.